fe/src/main/java/org/apache/doris/analysis/SetVar.java

```diff
@@ -102,6 +103,18 @@ public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
             value = new StringLiteral(((SlotRef) value).getColumnName());
         }
 
+        if (variable.equalsIgnoreCase(SessionVariable.SQL_MODE)) {
```

Comment:
I think it is better to move this logic to line 128

Reply:
why? the value will be assigned to result, so I substitute the value before the result assigned.